### PR TITLE
First step learning graphql

### PR DESCRIPTION
**Pull request by frudisch**
Added simple in-memory database storing video entries and updated graphql interface accordingly

### CODE REVIEW

File: src/test/kotlin/vibroad/service/graphql/VideoSelectionIntegrationTest.kt [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+@MicronautTest
+class VideoSelectionIntegrationTest(
+        private val graphQlTestClient: GraphQlTestClient
+) : StringSpec({
+
+    "should return a array containing the single stored video entity" {
+        val result = graphQlTestClient.sendQuery("select_all_videos.txt")
+
+        println(result)
```
Comment (si87) [inline src/test/kotlin/vibroad/service/graphql/VideoSelectionIntegrationTest.kt:17]: what is that? :P

File: src/test/kotlin/vibroad/service/graphql/VideoSelectionIntegrationTest.kt [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        }
+    }
+
+    "should return the video identified by the passed id in the query" {
+        val result = graphQlTestClient.sendQuery("select_video_by_id.txt")
+
+        println(result)
+
+        expectThat(result.get("data").get("video")) {
+            get { get("id").textValue() }.isEqualTo("b670c9fe-ea3e-493f-b7f5-9186e640555e")
```
Comment (si87) [inline src/test/kotlin/vibroad/service/graphql/VideoSelectionIntegrationTest.kt:32]: ok for now, great use case for custom matchers

**Review by si87 — CHANGES_REQUESTED** [review]

File: src/main/kotlin/vibroad/service/graphql/DataFetcher.kt [hunk 3]
```diff
@@ -0,0 +1,35 @@
+package vibroad.service.graphql
+
+import graphql.schema.DataFetcher
+import graphql.schema.DataFetchingEnvironment
+import vibroad.service.model.Video
+import vibroad.service.service.VideoService
+import java.util.*
```
[the remark this answers is not in the document]
Comment (frudisch) [reply src/main/kotlin/vibroad/service/graphql/DataFetcher.kt:7]: an der Stelle geht es um die UUID, gibt es da ein kotlin equivalent ?

File: src/test/kotlin/vibroad/service/graphql/VideoSelectionIntegrationTest.kt (hunk 1 above)
Comment (frudisch) [reply src/test/kotlin/vibroad/service/graphql/VideoSelectionIntegrationTest.kt:17]: debugging ... was ist der standard in kotlin / micronaut ? log.info(...) oder println ?

File: src/test/kotlin/vibroad/service/graphql/VideoSelectionIntegrationTest.kt (hunk 2 above)
Comment (frudisch) [reply src/test/kotlin/vibroad/service/graphql/VideoSelectionIntegrationTest.kt:32]: i tried something like that but failed badly ... need to gather more experience with kotlin

File: src/main/kotlin/vibroad/service/model/Video.kt [hunk 4]
```diff
@@ -0,0 +1,14 @@
+package vibroad.service.model
+
+import java.util.*
+
+class Video(
```
[the remark this answers is not in the document]
Comment (frudisch) [reply src/main/kotlin/vibroad/service/model/Video.kt:5]: done

File: src/main/kotlin/vibroad/service/model/Video.kt [hunk 5]
```diff
@@ -0,0 +1,14 @@
+package vibroad.service.model
+
+import java.util.*
```
[the remark this answers is not in the document]
Comment (frudisch) [reply src/main/kotlin/vibroad/service/model/Video.kt:3]: used for UUID, is there another class in kotlin ? removed the * import ... wrong IDE configuration

File: src/main/kotlin/vibroad/service/graphql/DataFetcher.kt [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+    override fun get(env: DataFetchingEnvironment): Video? {
+        val id = UUID.fromString(env.getArgument<String>("id"))
+        return videoService.getById(id)
+    }
+}
+
+@Singleton
+class VideoByNameFetcher(
+        private val videoService: VideoService
+) : DataFetcher<List<Video?>> {
```
[the remark this answers is not in the document]
Comment (frudisch) [reply src/main/kotlin/vibroad/service/graphql/DataFetcher.kt:30]: yeah ... no result should return an empty list and not a null list ...

File: src/test/kotlin/vibroad/service/graphql/VideoSelectionIntegrationTest.kt (hunk 1 above)
Comment (si87) [reply src/test/kotlin/vibroad/service/graphql/VideoSelectionIntegrationTest.kt]: https://stackoverflow.com/questions/34416869/idiomatic-way-of-logging-in-kotlin

:D essentially using slf4j with Kotlin-Foo to retrieve a logger

**Review by si87 — APPROVED** [review]

**Pull request merged**